warehouse/templates/includes/file-details.html

```diff
@@ -118,26 +138,67 @@ <h3>Provenance</h3>
                 <li>Sigstore transparency entry: <a href="https://search.sigstore.dev/?logIndex={{ tlog_entry.logIndex }}">{{ tlog_entry.logIndex }}</a> </li>
                 <li>Sigstore integration time: {{ humanize(tlog_entry.integratedTime | int | ctime, time="true") }} </li>
               </ul>
-            </li>
+              {% set claims = attestation.certificate_claims %}
+              Source repository:
+              <ul>
+                <!-- 1.3.6.1.4.1.57264.1.12 | Source Repository URI-->
+                <!-- 1.3.6.1.4.1.57264.1.13 | Source Repository Digest-->
+                <li>Permalink: <a href="{{ repo_url_with_reference(claims.get("1.3.6.1.4.1.57264.1.12"), claims.get("1.3.6.1.4.1.57264.1.13")) }}">
+                    <code>{{ repository_name(bundle.publisher) }}#{{ claims.get("1.3.6.1.4.1.57264.1.13") }}</code>
+                </a></li>
+                <!-- 1.3.6.1.4.1.57264.1.14 | Source Repository Ref-->
+                <li>Branch / Tag: <a href="{{ repo_url_with_reference(claims.get("1.3.6.1.4.1.57264.1.12"), claims.get("1.3.6.1.4.1.57264.1.14")) }} ">
+                    <code>{{ claims.get("1.3.6.1.4.1.57264.1.14") }}</code>
+                </a></li>
+                <!-- 1.3.6.1.4.1.57264.1.15 | Source Repository Identifier-->
+                <!-- 1.3.6.1.4.1.57264.1.16 | Source Repository Owner URI-->
+                <li>Owner: <a href="{{ claims.get("1.3.6.1.4.1.57264.1.16") }}">{{ claims.get("1.3.6.1.4.1.57264.1.16") }}</a></li>
+                <!-- 1.3.6.1.4.1.57264.1.17 | Source Repository Owner Identifier-->
+                <!-- 1.3.6.1.4.1.57264.1.22 | Source Repository Visibility At Signing-->
+                <li>Access: <code>{{ claims.get("1.3.6.1.4.1.57264.1.22") }}</code></li>
+              </ul>
+              Build information:
+              <ul>
+                <!-- 1.3.6.1.4.1.57264.1.8 | Issuer-->
+                <!-- 1.3.6.1.4.1.57264.1.9 | Build Signer URI-->
+                <!-- 1.3.6.1.4.1.57264.1.10 | Build Signer Digest-->
+                <li>Token Issuer: <code>{{ claims.get("1.3.6.1.4.1.57264.1.8") }}</code></li>
+                <!-- 1.3.6.1.4.1.57264.1.11 | Runner Environment-->
+                <li>Build Environment: <code>{{ claims.get("1.3.6.1.4.1.57264.1.11") }}</code></li>
+                <!-- 1.3.6.1.4.1.57264.1.18 | Build Config URI-->
+                <!-- 1.3.6.1.4.1.57264.1.19 | Build Config Digest-->
+                <li>Build Configuration File:
+                  <a href="{{ workflow_url(bundle.publisher, claims) }}">
+                    <code>{{ workflow_filename(bundle.publisher) }}#{{ claims.get("1.3.6.1.4.1.57264.1.19") }}</code>
+                  </a>
+                </li>
+                <!-- 1.3.6.1.4.1.57264.1.20 | Build Trigger-->
+                <li>Trigger Event: <code>{{ claims.get("1.3.6.1.4.1.57264.1.20") }}</code></li>
+                <!-- 1.3.6.1.4.1.57264.1.21 | Run Invocation URI-->
+              </ul>
```

Comment:
Doing these mappings/selections within the template is likely going to be brittle/hard to test going forwards -- I think we'd be better off defining a Jinja2 function or other helper within the Python codebase to do these transforms.

(I'm not 100% sure how best to do this, since the way we render each claim varies based on its content/expected format, e.g. URLs. Maybe @di has thoughts?)

Reply:
The new version uses python helper functions - however, it still delegates the formatting to the view.

( NB: I quickly searched in the codebase to find an existing pattern to reuse, but I was not successful.)